packages/core/contracts/financial-templates/common/FundingRateApplier.sol

```diff
@@ -141,40 +142,37 @@ contract FundingRateApplier is Lockable {
             currentTime,
             paymentPeriod,
             _latestFundingRatePerSecond.rawValue,
-            effectiveFundingRateForPeriod.rawValue
+            periodRate.rawValue
         );
 
         lastUpdateTime = currentTime;
     }
 
     function _calculateEffectiveFundingRate(
         uint256 paymentPeriodSeconds,
-        FixedPoint.Unsigned memory fundingRatePerSecond,
-        FixedPoint.Unsigned memory feeMultiplier
-    ) internal pure returns (FixedPoint.Unsigned memory, FixedPoint.Unsigned memory) {
+        FixedPoint.Signed memory fundingRatePerSecond,
+        FixedPoint.Unsigned memory currentCumulativeFundingRateMultiplier
+    ) internal pure returns (FixedPoint.Unsigned memory, FixedPoint.Signed memory) {
         // Determine whether `fundingRatePerSecond` implies a negative or positive funding rate,
         // and apply it over a pay period.
-        FixedPoint.Unsigned memory ONE = FixedPoint.fromUnscaledUint(1);
-        FixedPoint.Unsigned memory effectiveFundingRateForPeriod = ONE;
-
-        // If `fundingRatePerSecond` == 1, then maintain the current multiplier.
-        FixedPoint.Unsigned memory newFeeMultiplier = feeMultiplier;
-        if (fundingRatePerSecond.isGreaterThan(ONE)) {
-            // If `fundingRatePerSecond` > 1, then first scale the funding over the pay period:
-            // (`fundingRatePerSecond` - 1) * payPeriod = effectiveFundingRate.
-            // Next, multiply the current multipier by (1 + effectiveFundingRate).
-
-            effectiveFundingRateForPeriod = ONE.add(fundingRatePerSecond.sub(ONE).mul(paymentPeriodSeconds));
-            newFeeMultiplier = feeMultiplier.mul(effectiveFundingRateForPeriod);
-        } else if (fundingRatePerSecond.isLessThan(ONE)) {
-            // If `fundingRatePerSecond` < 1, then first scale the funding over the pay period:
-            // (1 - `fundingRatePerSecond`) * payPeriod = effectiveFundingRate.
-            // Next, multiply the current multipier by (1 - effectiveFundingRate).
-
-            effectiveFundingRateForPeriod = ONE.sub(ONE.sub(fundingRatePerSecond).mul(paymentPeriodSeconds));
-            newFeeMultiplier = feeMultiplier.mul(effectiveFundingRateForPeriod);
-        }
-
-        return (newFeeMultiplier, effectiveFundingRateForPeriod);
+        FixedPoint.Signed memory ONE = FixedPoint.fromUnscaledInt(1);
+
+        // Multiply the per-second rate over the number of seconds that have elapsed to get the period rate.
+        FixedPoint.Signed memory periodRate = fundingRatePerSecond.mul(SafeCast.toInt256(paymentPeriodSeconds));
+
+        // Add one to create the multiplier to scale the existing fee multiplier.
+        FixedPoint.Signed memory signedPeriodMultiplier = ONE.add(periodRate);
+
+        // Max with 0 to ensure the multiplier isn't negative, then cast to an Unsigned.
```

Comment:
In which cases is it possible for the period rate to be negative?

Reply:
If the funding rate was negative and then it was multiplied by time s.t. the result is < -1.

Let's say the per-second funding rate were -0.1 (vv high) and the time that had passed was 20 seconds, the funding rate would be -2. After adding to 1, we would get -1. But you can't multiply sponsor's debt by a negative value (that would mean the token holders have a liability). Basically, the most you can do is send the cumulative multiplier to 0, which would be a period rate of -1, making the notional of each token 0 (each token has no value). So, I think it makes sense to bottom this value out at 0.

Reply:
Ahh yes I agree it makes sense to bottom the multiplier at 0